main.tf

```diff
@@ -0,0 +1,87 @@
+terraform {
+    backend "azurerm" {
+      resource_group_name   = "AmericanExpress2_MadhuRoy_ProjectExercise"
+      storage_account_name  = "madhustoreacc"
+      container_name        = "madhucontain"
+      key                   = "terraform.tfstate"
+    }
+    required_providers {
+        azurerm = {
+            source = "hashicorp/azurerm"
+            version = ">=2.49"
+        }
+    }
+}
+provider "azurerm" {
+    skip_provider_registration = true
+    features {}
+}
+data "azurerm_resource_group" "main" {
+    name = "AmericanExpress2_MadhuRoy_ProjectExercise"
+}
+
+resource "azurerm_app_service_plan" "main" {
+    name = "terraformed-asp"
+    location = var.location
+    resource_group_name = var.prefix_name
+    kind = "Linux"
+    reserved = true
+
+    sku {
+        tier = "Basic" 
+        size = "B1"
+    }
+}
+
+resource "azurerm_app_service" "main" {
+    name = "madhuterraformhelloapp"
+    location = var.location
+    resource_group_name = var.prefix_name
+    app_service_plan_id = azurerm_app_service_plan.main.id
+
+    site_config {
+    app_command_line = ""
+    linux_fx_version = "DOCKER|madhuaxp/todo-app:latest"
+    }
+
+    app_settings = {
+    "DOCKER_REGISTRY_SERVER_URL" = "https://index.docker.io"
+    "MONGO_CONNECTION_URL" = "mongodb://${azurerm_cosmosdb_account.maindbaccount.name}:${azurerm_cosmosdb_account.maindbaccount.primary_key}@${azurerm_cosmosdb_account.maindbaccount.name}.mongo.cosmos.azure.com:10255/DefaultDatabase?ssl=true&replicaSet=globaldb&retrywrites=false&maxIdleTimeMS=120000"
```

Comment:
Here you're not setting any of your other environment variables which are set on your app (like your client id, client secret etc.) - are you manually adding them in via the Portal? If so we should be also specifying those here & registering them as input variables - we can either provide defaults if they're not sensitive, or pass them through if they are